hudi-common/src/main/java/org/apache/hudi/common/fs/HoodieSerializableFileStatus.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.common.fs;
+
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.Path;
+
+import java.io.Serializable;
+
+/**
+ * A serializable wrapper for FileStatus.
+ * <p>
+ * Hadoop 2.x FileStatus does not implement Serializable and can cause issues. (HUDI-5936)
+ * This class is supposed to make sure FileStatus can be safely serialized by wrapping FileStatus
+ * with it, and it should be only used when we absolutely need to serialize FileStatus.
+ */
+public class HoodieSerializableFileStatus extends FileStatus implements Serializable {
+
+  Path path;
+  long length;
+  boolean isDirectory;
+  short blockReplication;
+  long blockSize;
```

Comment:
There is a `HoodieFileStatus`, can we use that?

Reply:
Good point, I just updated it to reuse `HoodieFileStatus`. It works well